crates/nu-command/src/system/benchmark.rs

```diff
@@ -41,12 +45,27 @@ impl Command for Benchmark {
         let redirect_stderr = call.redirect_stderr;
 
         let mut stack = stack.captures_to_stack(&capture_block.captures);
+
+        // In order to provide the pipeline as a positional, it must be converted into a value.
+        // But because pipelines do not have Clone, this one has to be cloned as a value
+        // and then converted back into a pipeline for eval_block().
+        // So, the metadata must be saved here and restored at that point.
```

Comment:
Not 100% sure I can follow your comment here? The following two lines are concerned with the input and not the closure in the positional.

Reply:
No, "the positional" means the positional inside the closure itself. `{|foo| ...}` ← `foo` is the positional. In order to provide the pipeline as a value for `foo` via `stack.add_var()`, it must be turned into `input_val`. `input_metadata` is also collected for reasons stated.